content/en/docs/components/pipelines/sdk/connect-api.md

```diff
@@ -4,227 +4,337 @@ description = "How to connect to Kubeflow Pipelines using the SDK client and con
+                    f"HTTP status code '{resp.status_code}' for GET against: {redirect_url_obj.geturl()}"
+                )
+
+            # set the login url
+            auth_session["dex_login_url"] = resp.url
+
+        ################
+        # Attempt Dex Login
+        ################
+        resp = s.post(
+            auth_session["dex_login_url"],
+            data={"login": username, "password": password},
+            allow_redirects=True
+        )
+        if len(resp.history) == 0:
+            raise RuntimeError(
+                f"Login credentials were probably invalid - "
+                f"No redirect after POST to: {auth_session['dex_login_url']}"
+            )
+
+        # store the session cookies in a "key1=value1; key2=value2" string
+        auth_session["session_cookie"] = "; ".join([f"{c.name}={c.value}" for c in s.cookies])
+
+    return auth_session
+```
 
-As mentioned above, the Kubeflow Pipelines API Kubernetes service is `ml-pipeline-ui`.
+This Python code uses the above `get_istio_auth_session()` function to instantiate a `kfp.Client()`:
+
+```python
+import kfp
+
+KUBEFLOW_ENDPOINT = "http://localhost:8080"
+KUBEFLOW_USERNAME = "user@example.com"
+KUBEFLOW_PASSWORD = "12341234"
+
+auth_session = get_istio_auth_session(
+    url=KUBEFLOW_ENDPOINT,
+    username=KUBEFLOW_USERNAME,
+    password=KUBEFLOW_PASSWORD
+)
+
+client = kfp.Client(host=f"{KUBEFLOW_ENDPOINT}/pipeline", cookies=auth_session["session_cookie"])
+print(client.list_experiments())
+```
 
-Using [Kubernetes standard mechanisms to discover the service](https://kubernetes.io/docs/concepts/services-networking/service/#discovering-services), you can access `ml-pipeline-ui` service from a Pod in the same namespace by DNS name:
+
+---
+
+
+## Connect to Kubeflow Pipelines from within the same cluster
+
+### Non-multi-user mode
+
+You can access `Service/ml-pipeline-ui` from a Pod in the same namespace with its [DNS name](https://kubernetes.io/docs/concepts/services-networking/service/#discovering-services):
 
 ```python
 import kfp
+
 client = kfp.Client(host='http://ml-pipeline-ui:80')
 print(client.list_experiments())
 ```
 
-Or, you can access `ml-pipeline-ui` service by using environment variables:
+Or, you can access `Service/ml-pipeline-ui` using the automatically generated environment variables:
 
 ```python
 import kfp
 import os
+
 host = os.getenv('ML_PIPELINE_UI_SERVICE_HOST')
 port = os.getenv('ML_PIPELINE_UI_SERVICE_PORT')
 client = kfp.Client(host=f'http://{host}:{port}')
 print(client.list_experiments())
 ```
 
-When accessing Kubeflow Pipelines from a Pod in a different namespace, you must access by the service name and the namespace:
+When accessing Kubeflow Pipelines from a Pod in a different namespace, you must explicitly specify the `kubeflow` namespace:
 
 ```python
 import kfp
-namespace = 'kubeflow' # or the namespace you deployed Kubeflow Pipelines
+
+namespace = 'kubeflow' # or the namespace in which you deployed Kubeflow
```

Comment:
To be explicit about what is deployed in this namespace:

```
# or the namespace in which you deployed Kubeflow Pipelines
```

Reply:
Changed.